### PR TITLE
feat: add AWS s3 as a helm repository backend

### PR DESCRIPTION
**Pull request by mrparkers**
this is a breaking change - the config options have changed and are no longer compatible with v1.

### CODE REVIEW

File: src/verify-conditions.js [hunk 1]
```diff
@@ -47,45 +48,105 @@ const verifyConditions = async (
         );
     }
 
-    // verify that the GITHUB_TOKEN env variable is set
-    // if it isn't set, throw immediately, since verification functions below won't work
-    if (!process.env.GITHUB_TOKEN) {
+    // verify that either `aws` or `github` config option is set with correct info
+    // throw immediately if both are set, or if neither are set
+    if ((!github && !aws) || (github && aws)) {
```
Comment (alexashley) [inline src/verify-conditions.js:53]: I don't know if this reads any better, but throwing it out there:

```js
if (Boolean(aws) === Boolean(github)) {

}
```

More succinctly `!aws` would work too (although it inverts the value).

**Review by alexashley — APPROVED** [review]
:shipit:

File: src/verify-conditions.js (hunk 1 above)
Comment (mrparkers) [reply src/verify-conditions.js:53]: thanks for the suggestion.  although I don't love the way it looks now, I think it's easier to understand

File: test/verify-conditions.spec.js [hunk 2]
```diff
@@ -29,118 +33,245 @@ describe("verify conditions", () => {
...
+        describe("when the AWS region is not specified", () => {
+            beforeEach(() => {
+                delete expectedPluginConfig.aws.region;
+            });
 
-        expect(getRepositoryBranch).toHaveBeenCalledWith(expectedRepoOwner, expectedRepoName, pluginConfig.githubPagesBranch);
-    });
+            it("should throw an error", async () => {
+                await expect(() => verifyConditions(expectedPluginConfig, context)).rejects.toThrow(AggregateError);
+                await expect(() => verifyConditions(expectedPluginConfig, context)).rejects.toThrow("Expected `aws.region`, `aws.bucket`, and `aws.bucketUrl` config options to be set");
```
Comment (ChrisSchreiber) [inline test/verify-conditions.spec.js:68]: Would it be worth it to define the the exception messages in a module and de-duplicate the strings? It always make me uncomfortable when tests depend on strings that might get changed to fix grammar or improve readability it logging.

**Review by ChrisSchreiber — APPROVED** [review]
One suggestion I don't feel strongly about. Looks great!

File: test/verify-conditions.spec.js (hunk 2 above)
Comment (mrparkers) [reply test/verify-conditions.spec.js:68]: I agree, but I think I'll follow up with this later so I can get this merged and released.

**Pull request merged**